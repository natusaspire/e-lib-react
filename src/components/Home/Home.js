import React, { PureComponent } from 'react';

import Hero from '@/components/Hero';
import heroImage from '@/assets/hero.jpeg';

const PAGE_TITLE = 'E-lib';

class Home extends PureComponent {
  componentDidMount() {
    document.title = PAGE_TITLE;
  }

  render() {
    return (
      <div className="main">
        <div className="main__hero">
          <Hero imageUrl={heroImage} />
        </div>
      </div>
    );
  }
}

export default Home;
