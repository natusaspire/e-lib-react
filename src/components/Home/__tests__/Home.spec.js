import React from 'react';
import { shallow } from 'enzyme';

import Home from '@/components/Home';

jest.mock('@/components/Hero', () => 'Hero');
jest.mock('@/assets/hero.jpeg', () => 'hero.jpeg');

describe('<Home />', () => {
  beforeEach(() => {
    global.document.title = '';
  });

  it('should render home content', () => {
    const wrapper = shallow(<Home />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should change document title on mount', () => {
    const title = 'E-lib';

    shallow(<Home />);

    expect(global.document.title).toBe(title);
  });
});
