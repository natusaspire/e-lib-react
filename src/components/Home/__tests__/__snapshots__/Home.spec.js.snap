// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render correctly 1`] = `
<div
  className="main"
>
  <div
    className="main__hero"
  >
    <Hero
      imageUrl="hero.jpeg"
    />
  </div>
</div>
`;
